chainer/training/extensions/evaluator.py

```diff
@@ -209,6 +228,10 @@ def evaluate(self):
 
         summary = reporter_module.DictSummary()
 
+        if self._progress_bar:
+            pbar = _IteratorProgressBar()
+            pbar.iterator = it
```

Comment:
How about specifying `it` in the constructor of `_IteratorProgressBar`, instead of assigning after creation?

Reply:
I'm not sure if we should do that.
Current implementation is simple and consistent with the implementation of `_TrainerProgressBar` in `progress_bar.py`.
And it's hard to pass `trainer` to `_TrainerProgressBar.__init__`. One option is making `trainer` as `property`, but it seems too complicated a bit to me.
What do you think? If you think it's not so important to be consistent, I will change as you suggested.

Reply:
I think consistency is not important here. `trainer` and `iterator` are specific to the respective classes and it's not the point we want to generalize.
(I agree the way `trainer` is passed to `_TrainerProgressBar` might not be optimal, but it's not easy to enhance. I think the current implementation is okay.)

As for whether to use `_IteratorProgressBar.__init__`, I think in principle if we *always* need something to initialize an instance, it should be passed to `__init__`.

Reply:
Okay. I fixed in 39f8c6f